src/sentry/api/serializers/models/integration.py

```diff
@@ -19,10 +19,78 @@ def serialize(self, obj, attrs, user):
             'provider': {
                 'key': provider.key,
                 'name': provider.name,
-            }
+            },
         }
 
 
+class IntegrationConfigSerializer(IntegrationSerializer):
+    def serialize(self, obj, attrs, user):
+        data = super(IntegrationConfigSerializer, self).serialize(obj, attrs, user)
+
+        data.update({
+            'config_organization': [],
+            'config_project': [],
+        })
+
+        try:
+            install = obj.get_installation()
+            data.update({
+                'config_organization': install.get_organization_config(),
+                'config_project': install.get_project_config(),
+            })
+        except NotImplementedError:
+            # The integration may not implement a Installed Integration object
+            # representation.
+            pass
+
+        return data
+
+
+@register(OrganizationIntegration)
+class OrganizationIntegrationSerializer(Serializer):
+    def get_attrs(self, item_list, user, *args, **kwargs):
+        # Lookup related project integrations
+        project_integrations = ProjectIntegration.objects \
+            .select_related('project') \
+            .filter(
+                integration_id__in=[i.integration_id for i in item_list],
+                project__organization_id__in=[i.organization_id for i in item_list],
+            )
+
+        project_integrations_by_org = defaultdict(list)
+        for pi in project_integrations:
+            project_integrations_by_org[pi.project.organization_id].append({
+                'project_id': pi.project_id,
+                'config_data': pi.config,
+            })
+
+        return {
+            i: {
+                'project_integrations': project_integrations_by_org.get(i.organization_id, [])
+            } for i in item_list
+        }
+
+    def serialize(self, obj, attrs, user, organization=None, project=None):
+        integration = serialize(obj.integration, user, IntegrationConfigSerializer())
```

Comment:
This os `O(n)` on calls to `get_organization_config`, which may very well contain API calls to determine configuration options. (JIRA will do this).

Reply:
can you add a comment saying that too?